Remove none-check from `publication_date`

https://github.com/letsbuilda/letsbuilda-pypi/blob/7bb06c0bd6f708a5d3f54d57430dffb5a6cb176a/src/letsbuilda/pypi/models/models_rss.py#L28
is causing mypy to say
```
src/letsbuilda/pypi/models/models_rss.py:37: error: Argument "publication_date" to "RSSPackageMetadata" has incompatible type "datetime | None"; expected "datetime"  [arg-type]
```

It looks like this field is always present, so lets just remove the none-check.

--- a/docs/source/changelog.rst
+++ b/docs/source/changelog.rst
@@ -1,6 +1,9 @@
 Changelog
 =========
 
+- :release:`5.0.2 <3rd February 2024>`
+- :bug:`74` Remove none-check from publication_date
+
 - :release:`5.0.1 <3rd February 2024>`
 - :bug:`67` Use a single function to get versioned package metadata
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,6 +1,6 @@
 [project]
 name = "letsbuilda-pypi"
-version = "5.0.1"
+version = "5.0.2"
 description = "A wrapper for PyPI's API and RSS feed"
 authors = [
     { name = "Bradley Reynolds", email = "bradley.reynolds@darbia.dev" },

--- a/src/letsbuilda/pypi/models/models_rss.py
+++ b/src/letsbuilda/pypi/models/models_rss.py
@@ -25,8 +25,6 @@ class RSSPackageMetadata:
         title = split_title[0]
         version = split_title[1] if len(split_title) == 2 else None  # noqa: PLR2004 - is not magic
 
-        publication_date = parsedate_to_datetime(data.get("pubDate")) if data.get("pubDate") is not None else None
-
         return cls(
             title=title,
             version=version,
@@ -34,5 +32,5 @@ class RSSPackageMetadata:
             guid=data.get("guid"),
             description=data.get("description"),
             author=data.get("author"),
-            publication_date=publication_date,  # type: ignore[arg-type]
+            publication_date=parsedate_to_datetime(data["pubDate"]),
         )

